hw/ip/rv_dm/data/rv_dm_testplan.hjson

```diff
@@ -463,6 +463,22 @@
       stage: V1
       tests: ["rv_dm_halt_resume_whereto"]
     }
+    {
+      name: progbuf_read_write_execute
+      desc: '''
+            Verify the functionality of program buffer by writing random data on program buffer.
+
+            - Program jtag_dmi_ral.dmcontrol.haltreq to 1'b1.
+            - Acknowledge haltreq by setting tl_mem_ral.halted to 0.
+            - Write random data on jtag_dmi_ral.pragram_buffer_* registers to access
+              program buffer.
+            - send a valid abstract command(AccessRegister) by setting postexec bit set.
+            - Read tl_mem_ral.program_buf_* registers to read what data is written
+              on program buffer.
```

Comment:
This is missing the important bit! We presumably have to check that the data we wrote is what we read back.